weblate/addons/models.py

```diff
@@ -194,6 +235,54 @@ class Meta:
         prefix = ""
 
 
+def execute_addon_event(
+    addon: Addon,
+    component: Component,
+    scope: Project | Component | str,
```

Comment:
What it really should be:

```suggestion
    scope: Translation | Component,
```

We don't need to go up here:

* If translation is passed, use translation
* Otherwise use component

It should match https://github.com/WeblateOrg/weblate/pull/10100/files#diff-815de81bffc610b75def2de884c550297811d0f3b4ec3ceb8538c9d261574aa5R296-R297